compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1106,11 +1106,97 @@ static LogicalResult setTransposeLikeOpRootConfig(func::FuncOp entryPointFn,
+    func::FuncOp entryPointFn, linalg::GenericOp genericOp) {
+  if (getLoweringConfig(genericOp)) {
+    return success();
+  }
+
+  unsigned numLoops = genericOp.getNumLoops();
+  if (numLoops == 0) return success();
+  if (!linalg::isElementwise(genericOp)) return success();
+
+  // Set the flow level tiling to the default.
+  SmallVector<int64_t> minTileSizes =
+      getMinTilingSizesForEachDim(entryPointFn, genericOp);
+  SmallVector<int64_t> maxTileSizes(numLoops, defaultWorkgroupTileSize);
+  SmallVector<int64_t> flowTileSizes =
+      getDefaultDistributedLevelTileSizes(genericOp, minTileSizes, maxTileSizes,
+                                          /*allowIncompleteTile=*/true);
+
+  // Adjust the number of workload per workgroup to at least 4096.
+  constexpr int64_t kMinimumWorkload = 4096;
+  auto shape = genericOp.getStaticLoopRanges();
+  int64_t numWorkload = 1;
+  for (auto en : llvm::enumerate(shape)) {
+    int64_t size = en.value();
+    if (size == ShapedType::kDynamicSize) {
+      numWorkload = ShapedType::kDynamicSize;
+      break;
+    }
+    int index = en.index();
+    if (flowTileSizes[index]) {
+      size = flowTileSizes[index];
+    }
+    numWorkload *= size;
+  }
+  for (unsigned currDim = 0;
+       numWorkload < kMinimumWorkload && currDim < numLoops;) {
+    int64_t currSize = flowTileSizes[currDim];
+    if (currSize == shape[currDim] || currSize == 0 ||
+        shape[currDim] == ShapedType::kDynamicSize ||
+        numWorkload == ShapedType::kDynamicSize) {
+      currDim++;
+      continue;
+    }
+    int64_t newSize = std::min<int64_t>(currSize * 2, shape[currDim]);
+    numWorkload = numWorkload / currSize * newSize;
+    flowTileSizes[currDim] = newSize;
+  }
+
+  // Adjust tiling sizes of vector levels to avoid large unroll factors.
+  SmallVector<int64_t> vecTileSizes(minTileSizes.begin(), minTileSizes.end());
+  for (auto operand : genericOp.getOutputOperands()) {
+    constexpr int64_t kMaxUnrollFactor = 8;
+    AffineMap map = genericOp.getTiedIndexingMap(operand);
+    int64_t vecSize = getVectorSize(entryPointFn, operand->get().getType());
+    int64_t currSize = 1;
+    for (auto dimExpr : llvm::reverse(map.getResults().drop_back())) {
+      unsigned pos = dimExpr.cast<AffineDimExpr>().getPosition();
+      if (vecTileSizes[pos] * currSize > vecSize * kMaxUnrollFactor) {
+        vecTileSizes[pos] = 1;
+        currSize = vecSize * kMaxUnrollFactor;
+      }
+    }
+    int fastestPos =
+        map.getResults().back().cast<AffineDimExpr>().getPosition();
+    vecTileSizes[fastestPos] =
+        std::min<int64_t>(vecTileSizes[fastestPos], kMaxUnrollFactor);
+  }
+
+  // Setting reduction tile sizes is a workaround to kick in peeling transform.
+  // The tiling won't happen because the sizes are zeros.
+  SmallVector<int64_t> zeros(numLoops, 0);
+
+  TileSizesListType tileSizes;
+  tileSizes.push_back(flowTileSizes);
+  tileSizes.push_back(vecTileSizes);
+  tileSizes.push_back(zeros);
+
```

Comment:
I think we should start printing debug information (`LLVM_DEBUG(llvm::dbgs() << ...`) dumping the resulting values of the computation. That would be very helpful for debugging.

Reply:
Good point! I think we can do it at the end of configuration. I'll work on it in a new PR.